Makefile

```diff
@@ -25,6 +25,7 @@ export CGO_CFLAGS
 TOOLS = \
 	github.com/fzipp/gocyclo \
 	gitlab.com/opennota/check/cmd/varcheck \
+	github.com/spacemonkeygo/openssl \
```

Comment:
The dependencies should go into the `vendor` folder, as part of the vendored dependencies. I have to admit that since we recently switched to go modules support, I'm not 100% up to date on how this process works anymore. Perhaps @oleorhagen could help with this part.

Reply:
Sure thing. These tools mentioned in the Makefile are for use in the Makefile only, not as part of the source tree.

@torbenh3, are you familiar with go modules?

Reply:
@oleorhagen  i am not familiar with go modules. This is actually my first change ever to go code.

Reply:
Alright, then all the more impressive!

If you wan't to have a crack at this yourself, then read https://blog.golang.org/using-go-modules, it should give you most of what you need :)

If not, I can do it for you, or alternatively spell out the commands you need. It is up to you :)

Reply:
@oleorhagen i would be happy if you pointed me at the things i need to do.
i have created a softlink to my git checkout at  /home/torbenh/go/src/github.com/mendersoftware/mender

Now the whole "go test" and "go install" does not find logrus...

main.go:22:2: cannot find package "github.com/sirupsen/logrus" in any of:
        /usr/lib/go-1.11/src/github.com/sirupsen/logrus (from $GOROOT)
        /home/torbenh/go/src/github.com/sirupsen/logrus (from $GOPATH)
make: *** [Makefile:185: govet] Error 1

i dont understand, how to make go search in the vendor path.
this is debian buster with:

go version go1.11.6 linux/amd64

Reply:
Alright, let's do this. Seems the mender-client repository is not fully migrated to go modules yet though actually, so we'll fall back to govendor.

```bash
# Get the govendor binary
go get github.com/kardianos/govendor
# Download the source for the ssl wrapper (Go) -- Probably not necessary if you've done so already, but it is idempotent, so no harm done
go get github.com/spacemonkeygo/openssl
# Add the new source to the vendor folder
govendor add github.com/spacemonkeygo/openssl
```

And now, commit the sources which have been added to the vendor folder by `govendor`, and the changes done to the `vendor/vendor.json` file.

To test it, use the makefile, as opposed to the regular go build and go test commands.

So - Build:
`make build`
Test:
`make test`

Let's see how this goes.

If you still get the same error as above, report your values of `GOPATH` and `GOROOT`, as well as a description of your source tree, go has some quirks for how it handles the go source files.

Reply:
Thanks for the advice. This has worked.

The softlink confused go. I had cd into the gopath. when pwd is below the GOPATH it works.